### DISCUSSION TITLE
How to reduce integration points

### DISCUSSION THREAD

**Discussion by MScKarl** [idaholab/moose#28210] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi!

I am interested in investigating the impact of Gauss points in my simulation. To do this I need to change the number of integration points in a 2D simulation, with  QUAD4 elements. Is the code below the way to do it, and how do I know the number of Gauss points?

```
[./Quadrature]
    type = GAUSS_LOBATTO 
    order = FIFTH
  [../]
```

If its not, is there an easy way to set the number of integration points in the MOOSE code?

Best regards,
Karl

**Comment by GiudGiud** [comment]
Hello

You ll have to look at the quadrature definition in libmesh
Setting the type and order is the way to do this in moose

Guillaume

**Reply by MScKarl** [reply]
Hi @GiudGiud!

Thank you for your answer.

Where can I access this in my moose program? I've found this link:

https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1QMonomial.html

But I don't quite understand it. It looks like some sort of header file in C programming (which I understand MOOSE is written in)?

Is there some instruction page with steps to follow or something? This to show where the correct header file is, and where to change?

/Karl

**Reply by GiudGiud** [reply]
yes pick a quadrature from this page
https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1QBase.html

then look at the source code, in C++ or the header.
There is no instruction page afaik. You may find something in the libmesh theory documentation, I've never looked there

> and where to change?

you already know where to change in the input file, this is the block you were modifying in MOOSE

**Reply by GiudGiud** [reply]
otherwise there is this page that I wrote a year ago

https://mooseframework.inl.gov/syntax/Executioner/Quadrature/index.html
